File: patterns/java/src/refactoring_guru/patterns/facade/example/some_media_library/BitrateReader.java
package refactoring_guru.patterns.facade.example.some_media_library;

public class BitrateReader {
    public static VideoFile read(VideoFile file, Codec codec) {
        // analysis
        // reading
        // some operation
        return file;
    }

    public static VideoFile convert(VideoFile buffer, Codec codec) {
        // analysis
        // some operation
        // converting
        return buffer;
    }
}

File: patterns/cs/src/RefactoringGuru/Facade/Example/SomeComplexMediaLibrary/BitrateReader.cs
using System;

namespace RefactoringGuru.Facade.Example.SomeComplexMediaLibrary
{
    class BitrateReader
    {
        public static VideoFile Read(VideoFile file, ICodec codec)
        {
            Console.WriteLine("BitrateReader: reading file...");
            return file;
        }

        public static VideoFile Convert(VideoFile buffer, ICodec codec)
        {
            Console.WriteLine("BitrateReader: writing file...");
            return buffer;
        }
    }
}
